src/txn/txn_rollback_to_stable.c

```diff
@@ -199,7 +271,13 @@ __txn_abort_newer_row_leaf(
 
         if ((insert = WT_ROW_INSERT(page, rip)) != NULL)
             __txn_abort_newer_insert(session, insert, rollback_timestamp);
+
+        /* Check for updates in the update list that satisfy the given timestamp. */
+        if (WT_ROW_UPDATE(page, rip) == NULL && WT_ROW_INSERT(page, rip) == NULL)
+            WT_RET(__txn_abort_row_ondisk_kv(session, page, rip, rollback_timestamp));
```

Comment:
Could there be updates needing rollback, and the on-disk value *also* needs rollback?  I'm not clear why the `if` is needed around this call.

Reply:
Yes, the check is not required. But this can be used for optimizing for the cases where there is a stable update is present, so there is no need to abort the on-disk value even if it is written a newer version, as later in the reconciliation, that will be updated. I am doing this change in the next pull request.